pkg/userdata/generate/generate.go

```diff
@@ -0,0 +1,223 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package generate
+
+import (
+	"bytes"
+	stdlibx509 "crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"errors"
+	"math/rand"
+	"net"
+	"text/template"
+
+	"github.com/talos-systems/talos/pkg/crypto/x509"
+)
+
+// CertStrings holds the string representation of a certificate and key.
+type CertStrings struct {
+	Crt string
+	Key string
+}
+
+// Input holds info about certs, ips, and node type.
+type Input struct {
+	Certs         *Certs
+	MasterIPs     []string
+	Index         int
+	ClusterName   string
+	ServiceDomain string
+	PodNet        []string
+	ServiceNet    []string
+	Endpoints     string
+	KubeadmTokens *KubeadmTokens
+	TrustdInfo    *TrustdInfo
+}
+
+// Certs holds the base64 encoded keys and certificates.
+type Certs struct {
```

Comment:
Is there a reason we'd use a new struct here instead of `userdata.security`/`userdata.ossecurity`?

Reply:
Mostly just because this PR is a migration of code. I plan on refactoring in a follow-up PR.